es-app/src/MetaData.cpp

```diff
@@ -35,12 +35,12 @@ MetaDataDecl folderDecls[] = {
 	{"thumbnail",   MD_PATH,                "",                 false,      "thumbnail",            "enter path to thumbnail"},
 	{"video",       MD_PATH,                "",                 false,      "video",                "enter path to video"},
 	{"marquee",     MD_PATH,                "",                 false,      "marquee",              "enter path to marquee"},
-	{"rating",      MD_RATING,              "0.000000",         false,      "rating",               "enter rating"},
-	{"releasedate", MD_DATE,                "not-a-date-time",  false,      "release date",         "enter release date"},
-	{"developer",   MD_STRING,              "unknown",          false,      "developer",            "enter game developer"},
-	{"publisher",   MD_STRING,              "unknown",          false,      "publisher",            "enter game publisher"},
-	{"genre",       MD_STRING,              "unknown",          false,      "genre",                "enter game genre"},
-	{"players",     MD_INT,                 "1",                false,      "players",              "enter number of players"}
+	{"rating",      MD_RATING,              "",                 true,       "rating",               "enter rating"},
```

Comment:
Hide (statistics=true) some elements for editing and define default values when not defined in XML. 1970-01-02 was picked as 1970-01-01 is already taken for games without <releasedate/>: "not-a-date-time" will put 1970-01-01 which then is rendered as "unknown". As a folder most of the time has not releasdate I wanted it to render as blank, this is what 1970-01-02 serves for.

Reply:
I understand the decision, but I am not fully thrilled with the approach here. 1970-01-01 is, effectively, a null date. 1970-01-02 is an actual value that we are choosing to have a different behavior. It's pretty much begging to be caught in an edge case at some point in time when nobody else knows why it was here and what for :)
What other options could we have here, potentially even reusing 1970-01-01?
Also, happy new year!

Reply:
Thanks, hope you started well too.

I have considered to reuse 1970-01-01 in some form, but the crucial part is that the flag folder/game is only available in the Metadata / Metadatalist class (which is used by the Filedata class). For rendering it is then applied to Datetimecomponent class (setValue()) e.g. in Detailgamelistview class. In the setValue() is then the display format decided (in getDisplayString()). 

Which means to me one must either put render/display information in the Metadataclass and or overload the setValue() to allow a Game/Folder flag for reusing 1970-01-01 to be displayed as "" for folders. 

Neither path did not make sense to me. Thus I have chosen 1970-01-02.

On a folder the releasedate information can not be edited from within ES. An user must set it deliberately via gamlist editing.

I also assume there will be no game released on one of these two dates for any system which are also supported by ES.

However, if users want one of these dates displayed (instead of "unknown"/""), they could append a non zero time to the releasedate (again via gamelist.xml editing).

I made the reference/usage of 1970-01-02 more coherent, by referring to [the constant](https://github.com/RetroPie/EmulationStation/blob/c87b0b712c706164180844b5d062f25097fce8ab/es-app/src/MetaData.cpp#L31). The usage can be more easily reconstructed with grep or an IDE.

HTH
--

I found some corner cases which promted the user to "Save changes?" even when there are no changes (an efffect that the initial author assumed that any statistical (non editable) value are at the end of the MetaDataDecl (in Metadata.class). I fixed it to be more robust, when statistical values [are in between](https://github.com/RetroPie/EmulationStation/blob/c87b0b712c706164180844b5d062f25097fce8ab/es-app/src/guis/GuiMetaDataEd.cpp#L270).

I also noted that the releasedate in die ES editor rolls over at the end of the 32bit Unix-time-epoch. But that is a different story. ;-)

Reply:
Thanks!